ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -123,11 +132,47 @@ public void write(OutputStream out) {
   }
 
   private void write(OutputStream output, GenericRow row) throws IOException {
-    objectMapper.writeValue(output, StreamedRow.row(row));
+    objectMapper.writeValue(output, StreamedRow.row(updateRow(row)));
     output.write("\n".getBytes(StandardCharsets.UTF_8));
     output.flush();
   }
 
+  private GenericRow updateRow(GenericRow genericRow) {
+    List<Object> columns = new ArrayList<>();
+    genericRow.getColumns()
+        .forEach(col -> columns.add(getValidObject(col)));
+    return new GenericRow(columns);
+  }
+
+  private Object getValidObject(Object object) {
```

Comment:
Can it be static?

Reply:
Hand crafted serialization code again. This ain't nice code.  Why are we re-writing values as strings before serializing? What problem are you trying to solve?

Reply:
The StreamsRow chocks when fields in GenericRow are complex objects such as struct. Struct isn't a jackson-serializable data type. For now we can send the string value of it but we need to figure out a better way to serialize the struct and other multi level nested fields with struct fields.

Reply:
> Struct isn't a jackson-serializable data type

I thought that might be the reason for this.  Rather than writing handcrafted serialization code, we should configure the Jackson ObjectMapper with a suitable serializer for `Struct`.

If you're not familiar with this I can help. (Though its been a while).